lib/model-interfaces/bedrock-agents/functions/request-handler/index.py

```diff
@@ -0,0 +1,163 @@
+import os
+import json
+import uuid
+from datetime import datetime
+from aws_lambda_powertools import Logger, Tracer
+from aws_lambda_powertools.utilities import parameters
+from aws_lambda_powertools.utilities.batch import BatchProcessor, EventType
+from aws_lambda_powertools.utilities.batch.exceptions import BatchProcessingError
+from aws_lambda_powertools.utilities.data_classes.sqs_event import SQSRecord
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from adapters.bedrock import BedrockAgent, AgentInputOutputAdapter
+
+from genai_core.utils.websocket import send_to_client
+from genai_core.types import ChatbotAction
+
+processor = BatchProcessor(event_type=EventType.SQS)
+tracer = Tracer()
+logger = Logger()
+
+AWS_REGION = os.environ["AWS_REGION"]
+API_KEYS_SECRETS_ARN = os.environ["API_KEYS_SECRETS_ARN"]
+CONFIG_PARAMETER_NAME = os.environ["CONFIG_PARAMETER_NAME"]
+
+sequence_number = 0
+config = json.loads(parameters.get_parameter(CONFIG_PARAMETER_NAME))
+
+
+def handle_run(record):
+    user_id = record["userId"]
+    data = record["data"]
+    agent_id = data["agentId"]
+    prompt = data["text"]
+    session_id = data.get("sessionId")
+
+    if not session_id:
+        session_id = str(uuid.uuid4())
+
+    # get the adapter from the registry
+
+    # create an agent adapter to invoke a Bedrock Agent using agentId and agentAliasId
+    agent = BedrockAgent(
+        agent_id=agent_id,
+        user_id=user_id,
+        session_id=session_id,
+        region_name=config.get("bedrock", {}).get("region", AWS_REGION),
+    )
+    # call the agent
+    response = agent.run(
+        prompt,
+    )
+
+    logger.info(response)
+    sequence_number = 0
+    run_id = str(uuid.uuid4())
+    for r in response:
+        # this is specific to Bedrock agents, would need a way to generalize
+        # it if we are to introduce other agents
+        if "chunk" in list(r.keys()):
+            send_to_client(
+                {
+                    "type": "text",
+                    "action": ChatbotAction.FINAL_RESPONSE.value,
```

Comment:
Will this be the final response? Doc suggests the chunk contains a part of an agent response.